### PR TITLE
feat(dropdown): create working dropdown

### PR DESCRIPTION
**Pull request by wesleycho**
- Create working dropdown component

Working Plunker [here](http://plnkr.co/edit/blhYNtxDtZxUFB4IQdsf?p=preview).

Implements #4

### CODE REVIEW

File: src/components/dropdown.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  </div>`
+})
+export class UibDropdown {
+  private open: boolean;
+}
+
+@Component({
+  selector: 'uib-dropdown-menu',
+  host: {
+    class: 'dropdown-menu'
```
Comment (Foxandxss) [inline src/components/dropdown.ts]: TS cries with this for me on the accordion. We will see when I merge the tooling.

File: src/components/dropdown.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+@Component({
+  selector: 'uib-dropdown',
+  properties: [
+    'open'
+  ]
+})
+@View({
+  directives: [NgClass],
+  template: `<div class="dropdown" [ng-class]="{open: open}">
```
Comment (Foxandxss) [inline src/components/dropdown.ts]: Did you try `host` in here? I think you had but you had issues.

File: src/components/dropdown.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/// <reference path="../../typings/tsd.d.ts" />
```
Comment (Foxandxss) [inline src/components/dropdown.ts]: No bueno.

File: src/components/dropdown.ts (hunk 2 above)
Comment (wesleycho) [inline src/components/dropdown.ts]: Pawel had a suggestion in the Angular repo when I mentioned not being able to use directives - I need to try
```js
properties: [
  '[class.open]: open'
]
```

Comment (wesleycho) [inline src/components/dropdown.ts]: Err, I had that suggestion wrong - [here](http://plnkr.co/edit/duj7JGoglJfX4b5oHK6g?p=preview) is a working implementation with changes though. This seems to have CSS ramifications though.

File: src/components/dropdown.ts (hunk 1 above)
Comment (wesleycho) [inline src/components/dropdown.ts]: Seems to work fine

File: src/components/dropdown.ts (hunk 2 above)
Comment (pkozlowski-opensource) [inline src/components/dropdown.ts]: I would definitively try `<div [class.open]="open">` here

File: src/components/dropdown.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import {Component, View} from 'angular2/angular2';
+
+@Component({
+  selector: 'uib-dropdown',
```
Comment (pkozlowski-opensource) [inline src/components/dropdown.ts]: Did we decide in the prefix already? I was under the impression that we want to cut ties to angular-ui, no?

File: src/components/dropdown.ts (hunk 2 above)
Comment (wesleycho) [inline src/components/dropdown.ts]: Already updated with that - see the current diff

File: src/components/dropdown.ts (hunk 4 above)
Comment (Foxandxss) [inline src/components/dropdown.ts]: Open to ideas, but not any ng- related.

Comment (wesleycho) [inline src/components/dropdown.ts]: @Foxandxss was against the `ng-` suggestion - we should probably hash this out in the meeting shortly. I would prefer `ng-bs-` myself.

File: src/components/dropdown.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import {Component, View} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-dropdown',
+  properties: [
+    'open'
+  ]
+})
+@View({
+  template: `<div class="dropdown" [class.open]="open">
```
Comment (Foxandxss) [inline src/components/dropdown.ts]: nitpicking, isn't better to put the template itself in a new line? @robwormald

Comment (Foxandxss) [inline src/components/dropdown.ts]: Like you said to rob the other day, it is easier for the eye :P

**Comment by Foxandxss** [comment]
I give you the honors to push the very first angular 2 bootstrap.

**Pull request closed without merging**